src/chain.h

```diff
@@ -403,6 +411,7 @@ class CDiskBlockIndex : public CBlockIndex
         READWRITE(hashPrev);
         READWRITE(hashMerkleRoot);
         READWRITE(nTime);
+        READWRITE(block_height);
```

Comment:
I think you need the condition here too for tests to pass, and I also think you need a global like in https://github.com/ElementsProject/elements/pull/433
```suggestion
if (g_block_height_in_header) {
        READWRITE(block_height);
}
```

Reply:
Again, shouldn't this be conditional?

Reply:
Doesn't matter if we don't serialize it... I think.

Reply:
We can make sure it doesn't matter by running all the tests with self.chain = 'regtest' here: https://github.com/ElementsProject/elements/blob/elements-0.17/test/functional/test_framework/test_framework.py#L85

Reply:
sorry why would this test it?

Reply:
I don't understand why this doesn't matter. Bitcoin block indexes sure won't have height here...

Reply:
Because unlike regtest2 (or any other custom chain) regtest doesn't default to true for the new option. Another qay to test it, is running all tests with the new option set to false (or defaulting to false).